src/schematools/importer/ndjson.py

```diff
@@ -172,7 +172,7 @@ def _format_db_value(self, field: DatasetFieldSchema, value) -> Any:
         elif field.id in self.inactive_relation_info:
             # Convert nested object to JSON string
             return json.dumps(value)
-        elif isinstance(value, (dict, list)):
+        elif isinstance(value, (dict, list)) and not field.is_json_object:
```

Comment:
This passes lists through, is that intended?

Reply:
That is intended, however writing it like this is not really clear. I changed it so that the `is_json_object` will become a separete case in the if/elif/else.